src/main/java/org/datadog/jmxfetch/JMXAttribute.java

```diff
@@ -351,6 +355,89 @@ MBeanAttributeInfo getAttribute() {
         return attribute;
     }
 
+    public ObjectName getBeanName() {
+      return beanName;
+    }
+
+    /**
+     * Get attribute alias.
+     *
+     * In order, tries to:
+     * * Use `alias_match` to generate an alias with a regular expression
+     * * Use `alias` directly
+     * * Create an generic alias prefixed with user's `metric_prefix` preference or default to `jmx`
+     *
+     * Argument(s):
+     * * (Optional) `field`
+     *   `Null` for `JMXSimpleAttribute`.
+     */
+    protected String getAlias(String field) {
+        String alias = null;
+
+        Filter include = getMatchingConf().getInclude();
+        LinkedHashMap<String, Object> conf = getMatchingConf().getConf();
+
+        String fullAttributeName =(field!=null)?(getAttribute().getName() + "." + field):(getAttribute().getName());
+
+        if (include.getAttribute() instanceof LinkedHashMap<?, ?>) {
+            LinkedHashMap<String, LinkedHashMap<String, String>> attribute = (LinkedHashMap<String, LinkedHashMap<String, String>>) (include.getAttribute());
+            if (attribute.get(fullAttributeName).get("alias_match") != null) {
+              Pattern p = Pattern.compile(attribute.get(fullAttributeName).get("alias_match"));
+              Matcher m = p.matcher(getBeanName().getCanonicalName() + "." + fullAttributeName);
```

Comment:
My thinking was that as the alias_match was defined inside the attribute yaml config it should include it in the match statement, and it also removed some extra duplicity where you would have to append ".#attribute#" to the end of the alias config line to reproduce the full metric name.

Of course using this method if you didn't want the same attribute name you'd have to add it to the alias_match regex, so it's a similar issue either way.